QModManager/API/ModLoading/Internal/QMod.cs

```diff
@@ -0,0 +1,339 @@
+﻿namespace QModManager.API.ModLoading.Internal
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using Oculus.Newtonsoft.Json;
+    using QModManager.DataStructures;
+    using QModManager.Utility;
+
+    [JsonObject(MemberSerialization.OptIn)]
+    internal class QMod : IQMod, IQModSerialiable, IQModLoadable, ISortable<string>
```

Comment:
There is a lot going on in this class:  initializing from JSON, from a DLL, as an invalid mod with a name only, and it includes the `QModManager` singleton.  

I would like to see about refactoring all the common code into an abstract `QModBase` class so as to better separate the methods used for loading the mod from these different sources.

Thoughts?

Reply:
Pending